code/modules/projectiles/guns/energy.dm

```diff
@@ -58,6 +58,8 @@
 
 /obj/item/gun/energy/proc/update_ammo_types()
 	var/obj/item/ammo_casing/energy/shot
+	if(select > length(ammo_type)) /// If we decrease ammo types while selecting a removed one, we want to make sure it doesnt try to select an out of bounds index
+		select = 1
```

Comment:
nitpick: this is a regular code coment, not a doc-comment
```suggestion
	if(select > length(ammo_type)) // If we decrease ammo types while selecting a removed one, we want to make sure it doesnt try to select an out of bounds index
		select = 1
```

Separately: Since the goal here is to merely silence the runtime, and we don't really have a good idea of what mode to fall back to if this happens (i.e. we don't care which value `select` ends up taking), i think it'd be nicer to just

```suggestion
	// If we decrease ammo types while selecting a removed one, we want to make sure it doesnt try to select an out of bounds index
	select = clamp(select, 1, length(ammo_type))
```

simply because it's more compact.